### PR TITLE
Add the implementation for Vertex and Edge find

### PR DESCRIPTION
**Pull request by Incarnation-p-lee**
* As the graph is binding with database collection

* Find with Graph may not suitable here

Signed-off-by: Pan Li <panli@microsoft.com>

### CODE REVIEW

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            throw new UnexpectedGremlinSourceTypeException("should be the instance of GremlinSourceVertex");
+        }
+
+        final List<String> scriptList = new ArrayList<>();
+        final String id = source.getId();
+
+        Assert.notNull(id, "id should not be null");
+
+        scriptList.add(Constants.GREMLIN_PRIMITIVE_GRAPH);
+        scriptList.add(String.format(Constants.GREMLIN_PRIMITIVE_VERTEX, id));
```
Comment (yungezz) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java]: so only can find by id?

**Comment by Incarnation-p-lee** [comment]
typo here, should be **find** instead of insert

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java (hunk 1 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java]: As the client is binding with one collection, id is unique in a graph.

Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java]: Sorry for misunderstand, current we only implement findById, more will be added later, as signature may change then.

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptEdgeFindLiteral.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import com.microsoft.spring.data.gremlin.exception.UnexpectedGremlinSourceTypeException;
+import lombok.NoArgsConstructor;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@NoArgsConstructor
+public class GremlinScriptEdgeFindLiteral implements GremlinScript<String> {
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptEdgeFindLiteral.java]: Add some comments for this class？

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import com.microsoft.spring.data.gremlin.exception.UnexpectedGremlinSourceTypeException;
+import lombok.NoArgsConstructor;
+import org.springframework.lang.NonNull;
+import org.springframework.util.Assert;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@NoArgsConstructor
+public class GremlinScriptVertexFindLiteral implements GremlinScript<String> {
```
Comment (sophiaso) [inline src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java]: `GremlinScriptVertexFindLiteral`, does that mean this class is only used for vertex find query?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptEdgeFindLiteral.java (hunk 2 above)
Comment (sophiaso) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptEdgeFindLiteral.java]: Why do we separate edge operations and vertex operations into different classes? Is there any combined operations?

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java (hunk 3 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptVertexFindLiteral.java]: Yes, the query for find/insert is quit different.

File: src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptEdgeFindLiteral.java (hunk 2 above)
Comment (Incarnation-p-lee) [reply src/main/java/com/microsoft/spring/data/gremlin/conversion/GremlinScriptEdgeFindLiteral.java]: As you cannot query both edge and vertex at the same time. They all implement Script and generate the script for query. If put all together, we cannot restrict them by interface.

**Review by sophiaso — APPROVED** [review]

**Pull request merged**